community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/CommandStream.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.transaction;
+
+import java.io.IOException;
+
+import org.neo4j.helpers.collection.Visitor;
+import org.neo4j.kernel.impl.transaction.command.Command;
+
+/**
+ * A stream of commands from one or more transactions, that can be serialised to a transaction log or applied to a
+ * store.
+ */
+public interface CommandStream
```

Comment:
Can we please avoid using confusing names for things that we just introduce. Here we introduce a stream which is not a stream at all, i think we already have a wide range of confusing stuff in a code base.